2 create user endpoints

iczky opened the pull request:
finish issues #2 

there are some features I'm missing:
1.  Created_by reference schema
2.  Logout function

iczky committed:
Initialize a MongoDB connection
1. Create folder middleware
2. Init connection inside dbConnection-middleware.js
3. use the middleware after initialize express

Mongo connection #2

iczky committed:
Create users endpoint group
1. get all users
2. get user by id
3. update and delete the user
4. also create seed.js for testing purpose

user endpoint route group #2

iczky committed:
feat: add authentication using jwt
1. using passport to authenticate
2. implement login to return token
3. add auth middleware to user route

Close this issued #2

alvianzf commented on app.js:
```diff
@@ -1,40 +1,48 @@
 // error handler
-app.use(function(err, req, res, next) {
-  // set locals, only providing error in development
-  res.locals.message = err.message;
-  res.locals.error = req.app.get('env') === 'development' ? err : {};
-
-  // render the error page
-  res.status(err.status || 500);
-  res.render('error');
+// app.use(function (err, req, res, next) {
+//   // set locals, only providing error in development
+//   res.locals.message = err.message;
+//   res.locals.error = req.app.get("env") === "development" ? err : {};
+
+//   // render the error page
+//   res.status(err.status || 500);
+//   res.json({
+//     message: "test error",
+//     error: err,
+//   });
+// });
```
Please remove unnecessary codes and comments

alvianzf commented on middleware/authorization-middleware.js:
```diff
@@ -0,0 +1,38 @@
+        message: "Token is not available",
+      });
+    } else {
+      const token = authHeader.split(" ")[1];
```
Add a line before variable declaration to make it more readable.

iczky replied:
> According to @suchipi , Github member:
> 
> _**adding and removing line breaks is core to prettier's formatting algorithm, so preserving them isn't consistently possible.**_

Preserving line breaks for this use case is impossible according to prettier github contributor. Probably as right now i still use prettier for easier style code and indent, But im still trying other code config like ESlint

alvianzf replied:
understandable

alvianzf commented on routes/api/login/index.js:
```diff
@@ -0,0 +1,43 @@
+var bcrypt = require("bcrypt");
+var jwt = require("jsonwebtoken");
+
+const privateKey = "secret";
+
+router.post("/", async (req, res) => {
+  const { username, password } = req.body;
+
+  try {
+    const user = await User.findOne({ username });
+    const isPasswordCorrect = await bcrypt.compare(password, user.password);
+
+    if (isPasswordCorrect) {
+      const token = jwt.sign(
+        {
+          username: user.username,
+          id: user._id,
+          role: user.role,
+        },
+        privateKey
+      );
+
+      res.status(200).json({
+        message: "Login success",
+        data: token,
+      });
+    } else {
+      res.status(400).json({
+        message: "Login Failed",
+      });
+    }
+  } catch (error) {
+    res.status(400).json({
+      message: "username unavailable",
+      error: error,
+    });
+  }
+});
+
+module.exports = router;
```
Please change the group to `/auth`

login  --> `/auth/login`
logout--> `/auth/logout`

alvianzf commented on routes/api/logout/index.js:
```diff
@@ -0,0 +1,33 @@
+var express = require("express");
+var router = express.Router();
+var session = require("express-session");
+
+const app = express();
+
+app.use(
+  session({
+    secret: "secret",
+    resave: false,
+    saveUninitialized: true,
+  })
+);
+
+router.post("/", (req, res) => {
+  //clear session
+  console.log(req.session);
+  if (req.session) {
+    req.session.destroy((err) => {
+      if (err) {
+        res.status(400).send("Unable to log out");
+      } else {
+        res.send("Logout successful");
+      }
+    });
+  } else {
+    res.status(500).json({
+      message: "session unavailable",
+    });
+  }
+});
+
+module.exports = router;
```
Please change the group to /auth

login --> /auth/login
logout--> /auth/logout

alvianzf commented on routes/api/users/userManagement.js:
```diff
@@ -1,9 +1,77 @@
-router.get('/', function(req, res, next) {
-  res.send('respond with a resource');
+/* GET all users listing. */
+router.get("/users", async (req, res) => {
```
I need this to be
`/api/users`

not `/api/user/users`

alvianzf commented on routes/api/users/userManagement.js:
```diff
@@ -1,9 +1,77 @@
+});
+
+// get user based on id
+router.get("/user/:id", async (req, res) => {
```
this should be just `/:id`

alvianzf commented on seed.js:
```diff
@@ -0,0 +1,64 @@
+  }
+};
+
+// (async () => {
+//   console.log(await hashPassword("owner"));
+// })();
+
+// console.log(user);
+
+// console.log(hashPassword("manager"));
```
Remove unnecessary comments

alvianzf reviewed (changes requested):
Hi @iczky please fix the following tagged reviews

iczky committed:
fix: correction tagged reviews issues #2
1. Remove unnecessary codes
2. Change login and logout to auth route group
3. Fix users endpoint
4. Add getAll group route

Resolved this conversation #2

iczky committed:
fix: deleted unused file
1. deleted usermanagement.js

fix PR Review #2
